panoptes_aggregation/extractors/utilities.py

```diff
@@ -57,9 +58,93 @@ def pluck_fields(classification, pluck_keys):
             last_value = classification
             for keyi in key_path:
                 last_value = last_value[keyi.strip()]
-
-            answers["pluck." + key.strip()] = last_value
+            if(key == 'feedback'):
+                feedback_val = get_feedback_info(last_value)
+                if feedback_val is not None:
+                    answers['feedback'] = feedback_val
+            else:
+                answers["pluck." + key.strip()] = last_value
         except KeyError:
             continue
 
     return answers
+
+
+def get_feedback_info(feedback_dict):
+    '''
+    Extracts and processes classification success information from
+    the 'feedback' metadata
+
+    Inputs
+    ------
+    feedback_dict : list
+        List of dictionaries, where each entry correspond to
+        a feedback metadata in the classification JSON
+
+    Outputs
+    -------
+    feedback_data : dict
+        Dictionary containing information about
+        classification success, with the default keys being:
+            - 'success': which is a list of boolean success/failure keys
+                for each subtask in the classification
+            - 'agreement_score': The ratio of successful classifications
+                to the total number of subtasks.
+        and also the true values depending on the task type
+        (see `shape_tools.py`)
+
+    >>> feedback_dict = [{"id":100001,"answer":"3","success":True,"strategy":"singleAnswerQuestion",\
+            "failureEnabled":True,"failureMessage":"You got it wrong!",\
+            "successEnabled":True,"successMessage":"Congrats!"}]
+    >>> get_feedback_info(feedback_dict)
+    {'success': [True], 'true_answer': ['3'], 'agreement_score': 1.0}
+    '''
+
+    if not isinstance(feedback_dict, list):
+        return None
+
+    if len(feedback_dict) == 0:
+        return None
+
+    feedback_data = {}
+    feedback_data['success'] = []
+
+    key_list = FEEDBACK_STRATEGIES[feedback_dict[0]['strategy']]
+
+    for key in key_list:
+        feedback_data["true_" + key] = []
+        # feedback_data["user_"+key] = []
+
+    for classification in feedback_dict:
+        feedback_data['success'].append(classification['success'])
+        for key in key_list:
+            feedback_data["true_" + key].append(classification[key])
```

Comment:
You can clean this up a bit by using:

```python
for classification in feedback_dict:
    feedback_data['success'].setdefault('success', []).append(classification['success'])
    for key in key_list:
        feedback_data.setdefault(f'true_{key}', []).append(classification[key])
```

and remove line 110 and lines 114-116.

`setdefault` will use a default value only if the key does not already exist otherwise, it uses the existing key's value.